cmd/snap/cmd_quota.go

```diff
@@ -316,12 +444,47 @@ func (x *cmdQuotas) Execute(args []string) (err error) {
 			return fmt.Errorf("internal error: constraints is missing from daemon response")
 		}
 
-		constraintVal := "memory=" + strings.TrimSpace(fmtSize(int64(q.Constraints.Memory)))
-		currentVal := ""
-		if q.Current != nil && q.Current.Memory != 0 {
-			currentVal = "memory=" + strings.TrimSpace(fmtSize(int64(q.Current.Memory)))
+		var grpConstraints []string
+
+		// format memory constraint as memory=N
+		if q.Constraints.Memory != 0 {
+			grpConstraints = append(grpConstraints, "memory="+strings.TrimSpace(fmtSize(int64(q.Constraints.Memory))))
+		}
+
+		// format cpu constraint as cpu=NxM%,cpu-set=x,y,z
+		if q.Constraints.CPU != nil {
+			if q.Constraints.CPU.Count != 0 || q.Constraints.CPU.Percentage != 0 {
+				if q.Constraints.CPU.Count != 0 {
+					grpConstraints = append(grpConstraints, fmt.Sprintf("cpu=%dx", q.Constraints.CPU.Count))
+				}
+				if q.Constraints.CPU.Percentage != 0 {
+					grpConstraints = append(grpConstraints, fmt.Sprintf("cpu=%d%%", q.Constraints.CPU.Percentage))
+				}
+			}
+		}
+
+		if q.Constraints.CPUSet != nil && len(q.Constraints.CPUSet.CPUs) > 0 {
+			cpus := strutil.IntsToCommaSeparated(q.Constraints.CPUSet.CPUs)
+			grpConstraints = append(grpConstraints, "cpu-set="+cpus)
+		}
+
+		// format threads constraint as thread=N
+		if q.Constraints.Threads != 0 {
+			grpConstraints = append(grpConstraints, "thread="+strconv.Itoa(q.Constraints.Threads))
+		}
+
+		// format current resource values as memory=N,thread=N
+		var grpCurrent []string
+		if q.Current != nil {
+			if q.Current.Memory != 0 {
+				grpCurrent = append(grpCurrent, "memory="+strings.TrimSpace(fmtSize(int64(q.Current.Memory))))
+			}
+			if q.Constraints.Threads != 0 && q.Current.Threads != 0 {
```

Comment:
it not clear why you check `q.Constraints.Threads` here, could you please explain it in a comment?

Reply:
Good catch, it was an oversight, I'll get it fixed!